Swiftification/Array.swift

```diff
@@ -126,5 +126,15 @@ public extension Array {
     func take(numberOfElements: Int) -> Array {
         return Array(self[0..<max(min(numberOfElements, count), 0)])
     }
+    
+    /// Returns a Dictionary with keys generated by running each element through `map`.
+    @warn_unused_result
+    func toDictionary<K: Hashable>(map: Element -> K) -> [K: Element] {
```

Comment:
`groupBy` outputs an array of elements for each value instead of just the value. But, I guess I can already do this:

    Dictionary(array.map { (key($0), $0) })

So I'll just close this PR and do that instead.